sbpy/units/core.py

```diff
@@ -133,3 +136,150 @@ def spectral_density_vega(wfb):
+def reflectance(cross_section=None, reflectance=None, wfb=None, M_sun=None,
+        f_sun=None):
+    """Reflectance related equivalencies.
+
+    Support conversion from/to reflectance and scattering cross-section
+    to/from total flux and magnitude.
+
+    The default magnitude system is ``VEGAmag``, where the apparent magnitude
+    of Vega is assumed to be 0 in all and any wavelengths and bands.  If other
+    magnitude system is used, then it is implicitly inferred from the solar
+    magnitude passed by keyword 'M_sun'.
+
+    Parameters
+    ----------
+    cross_section : `astropy.units.Qauntity`
+        Total scattering cross-section
+    reflectance : `astropy.units.Quantity`
+        Average reflectance
+    wfb : `astropy.units.Quantity`, `synphot.SpectralElement`, string
+        Wavelength, frequency, or a bandpass of the corresponding flux
+        density being converted.  See
+        :func:`~synphot.SpectralElement.from_filter()` for possible
+        bandpass names.  ``wfb`` overrides ``f_sun`` and ``M_sun`` if
+        present at the same time.
+    f_sun : `astropy.units.Quantity`
+        Solar flux in a unit convertible to the unit of the quantity to be
+        converted.  If ``wfb`` is not provided, then ``f_sun`` will be used in
+        the conversion.  ``f_sun`` overrides ``M_sun`` if both present.
+    M_sun : `astropy.units.Quantity`
+        Solar magnitude in the same magnitude system as the quantity to be
+        converted.  If neither ``wfb`` nor ``f_sun`` is provided, then
+        ``M_sun`` will be used in the conversion.  If ``M_sun`` is not
+        present either, then the default V-band solar magnitude/flux will be
+        used.
+
+    Returns
+    -------
+    eqv : list
+        List of equivalencies
+
+    Examples
+    --------
+    >>> # Convertion between scattering cross-section and reflectance
+    >>> import numpy as np
+    >>> from astropy import units as u
+    >>> from sbpy.units import reflectance, VEGAmag
+    >>> mag = 3.4 * VEGAmag
+    >>> cross_sec = np.pi * (460 * u.km)**2
+    >>> ref = mag.to('1/sr', reflectance(cross_section=cross_sec))
+    >>> print('{0:.4f}'.format(ref))
+    0.0287 1 / sr
+    >>> mag1 = ref.to(VEGAmag, reflectance(cross_section=cross_sec))
+    >>> print('{0:.2f}'.format(mag1))
+    3.40 mag(VEGA)
+
+    >>> # Convertion bretween magnitude and scattering cross-section
+    >>> ref = 0.0287 / u.sr
+    >>> cross_sec = mag.to('km2', reflectance(reflectance=ref))
+    >>> radius = np.sqrt(cross_sec/np.pi)
+    >>> print('{0:.2f}'.format(radius))
+    459.69 km
+    >>> mag2 = cross_sec.to(VEGAmag, reflectance(reflectance=ref))
+    >>> print('{0:.2f}'.format(mag2))
+    3.40 mag(VEGA)
+    """
+    if wfb is not None:
+        try:
+            import synphot
+        except ImportError:
+            warn(AstropyWarning('synphot required for Vega-based magnitude'
+                            ' conversions.'))
+            return []
```

Comment:
The try...except is not needed since `sun.from_default` should raise an ImportError (either now or when my next PR is merged).